File: test/integration/docs_test.go
package integration_test

import (
	"path/filepath"
	"testing"

	"github.com/localstack/lstk/internal/must"
)

func TestDocsCommandGeneratesManPages(t *testing.T) {
	t.Parallel()
	dir := filepath.Join(t.TempDir(), "manpages")

	_, stderr, err := runLstk(t, testContext(t), "", testEnvWithHome(t.TempDir(), ""), "docs", "--format", "man", "--dir", dir)
	must.NoError(t, err, stderr)
	requireExitCode(t, 0, err)

	must.FileExists(t, filepath.Join(dir, "lstk.1"))
	must.FileExists(t, filepath.Join(dir, "lstk-start.1"))
	must.FileExists(t, filepath.Join(dir, "lstk-stop.1"))
}

func TestDocsCommandGeneratesMarkdown(t *testing.T) {
	t.Parallel()
	dir := filepath.Join(t.TempDir(), "markdown")

	_, stderr, err := runLstk(t, testContext(t), "", testEnvWithHome(t.TempDir(), ""), "docs", "--format", "markdown", "--dir", dir)
	must.NoError(t, err, stderr)
	requireExitCode(t, 0, err)

	must.FileExists(t, filepath.Join(dir, "lstk.md"))
	must.FileExists(t, filepath.Join(dir, "lstk_start.md"))
	must.FileExists(t, filepath.Join(dir, "lstk_stop.md"))
}

func TestDocsCommandRejectsInvalidFormat(t *testing.T) {
	t.Parallel()
	dir := t.TempDir()

	_, _, err := runLstk(t, testContext(t), "", testEnvWithHome(t.TempDir(), ""), "docs", "--format", "invalid", "--dir", dir)
	must.Error(t, err)
	requireExitCode(t, 1, err)
}

func TestDocsCommandIsHidden(t *testing.T) {
	t.Parallel()
	stdout, stderr, err := runLstk(t, testContext(t), t.TempDir(), testEnvWithHome(t.TempDir(), ""), "--help")
	must.NoError(t, err, stderr)

	must.NotContains(t, stdout, "docs")
}

File: test/e2e/tests/docs.test.ts
import { access } from "node:fs/promises";
import path from "node:path";
import { describe, expect, test } from "vitest";
import { lstk, tempHome } from "../support/index.ts";

// Ported from test/integration/docs_test.go.

async function fileExists(filePath: string): Promise<boolean> {
  try {
    await access(filePath);
    return true;
  } catch {
    return false;
  }
}

describe("lstk docs", () => {
  test("generates man pages", async () => {
    const home = await tempHome();
    const dir = path.join(home.path, "manpages");

    const run = await lstk(["docs", "--format", "man", "--dir", dir], { home });

    expect(run).toSucceed();
    // The generated man pages themselves are files on disk, not CLI output --
    // checked below by existence, not content. What belongs here is the
    // command's own stdout/stderr: a quiet success with nothing printed.
    expect(run.stdout).toPrintExactly("");
    expect(run.stderr).toPrintExactly("");
    expect(await fileExists(path.join(dir, "lstk.1"))).toBe(true);
    expect(await fileExists(path.join(dir, "lstk-start.1"))).toBe(true);
    expect(await fileExists(path.join(dir, "lstk-stop.1"))).toBe(true);
  });

  test("generates markdown", async () => {
    const home = await tempHome();
    const dir = path.join(home.path, "markdown");

    const run = await lstk(["docs", "--format", "markdown", "--dir", dir], { home });

    expect(run).toSucceed();
    // Same reasoning as the man-page test above: the generated markdown files
    // are checked by existence, not content; only the command's own
    // stdout/stderr is CLI output worth snapshotting.
    expect(run.stdout).toPrintExactly("");
    expect(run.stderr).toPrintExactly("");
    expect(await fileExists(path.join(dir, "lstk.md"))).toBe(true);
    expect(await fileExists(path.join(dir, "lstk_start.md"))).toBe(true);
    expect(await fileExists(path.join(dir, "lstk_stop.md"))).toBe(true);
  });

  test("rejects an invalid format", async () => {
    const home = await tempHome();
    const dir = path.join(home.path, "invalid");

    const run = await lstk(["docs", "--format", "invalid", "--dir", dir], { home });

    expect(run).toFail();
    expect(run.stderr).toPrintExactly("Error: unsupported format: invalid (use 'man' or 'markdown')");
  });

  test("is hidden from --help", async () => {
    const home = await tempHome();

    const run = await lstk(["--help"], { home });

    expect(run).toSucceed();
    // Not snapshotted: root --help lists every top-level command, so a full
    // snapshot here would flap on every unrelated command addition. The
    // point of this test is narrower -- "docs" specifically must not appear
    // -- which an absence substring check states directly.
    expect(run).not.toPrint("docs");
  });
});
